integrations/git_annex_test.go

```diff
@@ -0,0 +1,223 @@
+					remoteAnnexUUID, _, err := git.NewCommand(git.DefaultContext, "config", "remote.origin.annex-uuid").RunStdString(&git.RunOpts{Dir: repoPath})
+					remoteAnnexUUID = strings.TrimSpace(remoteAnnexUUID)
+
+					require.NoError(t, err)
+					require.Regexp(t,
+						regexp.MustCompile("^[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}$"),
+						remoteAnnexUUID,
+						"git annex sync should have been able to download the remote's annex uuid")
+
+					// Check that the file was uploaded
+					// - method 1: 'git annex whereis'
+					annexWhereis, _, err := git.NewCommand(git.DefaultContext, "annex", "whereis", "large.bin").RunStdString(&git.RunOpts{Dir: repoPath})
+
+					require.NoError(t, err)
+					require.True(t,
+					    strings.Contains(annexWhereis, " -- origin\n"),
+					    "git annex whereis should report the file is uploaded to origin")
+
+					// - method 2: look directly into the remote repo to find the file
+					// start by getting the key prefix of the uploaded file
+					// ref: https://git-annex.branchable.com/internals/hashing/
+					keyHashPrefix, _, err := git.NewCommand(git.DefaultContext, "annex", "examinekey", "--format=${hashdirlower}", annexKey).RunStdString(&git.RunOpts{Dir: repoPath})
+					require.NoError(t, err)
+
+					_, stat_err := os.Stat(path.Join(setting.RepoRootPath, API.Username, API.Reponame+".git", "annex", "objects", keyHashPrefix, annexKey, annexKey))
+					require.NoError(t, stat_err, "Annexed file should exist in remote .git/annex/objects folder")
+					// TODO: directly diff the source and target files
+				})
+
+				//fmt.Printf("Sleeping now. good luck.\n")
+				//time.Sleep(2 * time.Second) // DEBUG
+
+			})
+
+		})
+
+	})
+}
+
+func doGenerateRandomFile(size int, path string) (err error) {
+	// Generate random file
+	bufSize := 4 * 1024
+	if bufSize > size {
+		bufSize = size
+	}
+
+	buffer := make([]byte, bufSize)
+
+	f, err := os.Create(path)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	written := 0
+	for written < size {
+		n := size - written
+		if n > bufSize {
+			n = bufSize
+		}
+		_, err := rand.Read(buffer[:n])
+		if err != nil {
+			return err
+		}
+		n, err = f.Write(buffer[:n])
+		if err != nil {
+			return err
+		}
+		written += n
+	}
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func doCleanAnnexLockdown() {
+	// do chmod -R +w $REPOS in order to
+	// handle https://git-annex.branchable.com/internals/lockdown/
+	// > (The only bad consequence of this is that rm -rf .git doesn't work unless you first run chmod -R +w .git)
+	// If this isn't done, the test can only be run once, because it reuses its gitea-repositories/ path
+
+	filepath.WalkDir(setting.RepoRootPath, func(path string, d fs.DirEntry, err error) error {
+		if err == nil {
+			// 0222 == ugo+w, in octal unix permission notation
+			info, err := d.Info()
+			if err != nil {
+				return err
+			}
+
+			err = os.Chmod(path, info.Mode()|0222)
```

Comment:
Actually `chmod -R +w` is rather

```suggestion
			err = os.Chmod(path, info.Mode()|0200)
```